google-http-client/src/main/java/com/google/api/client/http/HttpRequest.java

```diff
@@ -932,7 +932,14 @@ public HttpResponse execute() throws IOException {
         } else {
           contentEncoding = encoding.getName();
           streamingContent = new HttpEncodingStreamingContent(streamingContent, encoding);
-          contentLength = contentRetrySupported ? IOUtils.computeLength(streamingContent) : -1;
+          if (contentRetrySupported) {
+            CachingByteArrayOutputStream outputStream = new CachingByteArrayOutputStream();
```

Comment:
IOUtils checks the size via streaming, and doesn't actually store the data in memory.  The same thing for sending the data.  A large object will currently never fully have to be in memory.

This change may cause an unexpected memory spike for users with large objects